2.segmentation/scripts/0.segment_images.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+#
+
+# In[1]:
+
+
+import pathlib
+
+import cv2
+import numpy as np
+
+# from segment_anything import SamPredictor
+import supervision as sv
+import supervision.draw.color as sv_color
+import torch
+from segment_anything import SamAutomaticMaskGenerator, sam_model_registry
+from skimage import io
+from tqdm import tqdm
+
+# ## Define Paths
+
+# In[2]:
+
+
+# import image_path
+image_path = pathlib.Path("../../data/0.raw_images/")
+# max_projection paths
+max_projection_path = pathlib.Path("../../data/1.maximum_projections/")
+# output mask path
+mask_path = pathlib.Path("../../data/2.masks/")
+# sqlite path
+sqlite_path = pathlib.Path("../../data/3.sqlite_output/")
+# models path
+models_path = pathlib.Path("../../data/models/")
+
+# create directories if they don't exist
+max_projection_path.mkdir(parents=True, exist_ok=True)
+mask_path.mkdir(parents=True, exist_ok=True)
+sqlite_path.mkdir(parents=True, exist_ok=True)
+models_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[3]:
+
+
+# Generate a list of all the images in the image_path directory of max projection images
+image_list = list(max_projection_path.glob("*.tif"))
+
+
+# ## Load in the SAM model
+
+# In[4]:
+
+
+DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+print(DEVICE)
+
+# # download the model if it doesn't exist
+# if not pathlib.Path("../../models/vit_h.pth").exists():
+#     !wget
+
+MODEL_TYPE = "vit_h"
+CHECKPOINT_PATH = "../../data/models/vit_h.pth"
+sam = sam_model_registry[MODEL_TYPE](checkpoint=CHECKPOINT_PATH).to(device=DEVICE)
+mask_generator = SamAutomaticMaskGenerator(sam)
+
+
+# ## Loop over the files and segment masks
+
+# In[5]:
+
+
+for image_path in tqdm(image_list):
```

Comment:
I might just be confused here, but why would you use image_path which is to the raw images to generate masks and not the max_projected images? I would assume that using the raw image paths would make a mask per z-slice. But I could be misunderstanding the code.